docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+# Contributing to the monorepo
+
+## Getting started
+
+- Install [Node.js](https://nodejs.org) version 14.
+  - If you're using [NVM](https://github.com/creationix/nvm#installation) (recommended), `nvm use` will ensure that the right version is installed.
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
+- Run `yarn install` to install dependencies and run any required post-install scripts.
+- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass linting before you push a branch.
+
+## Testing
+
+### Running unit tests
+
+- Run `yarn workspace <workspaceName> run test` to run all tests for a package.
+- Run `yarn workspace <workspaceName> run jest --no-coverage <file>` to run a test file within the context of a package.
+- Run `yarn test` to run tests for all packages.
+
+> **Note**  
+> `workspaceName` here is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+### Integrating with other projects
+
+When developing changes to packages within this repository that a different project depends upon, you may wish to load those changes into the project and test them locally or in CI before publishing proper releases of those packages. To solve that problem, this repository provides a mechanism to publish "preview" versions of packages to GitHub Package Registry. These versions can then be used in the project like any other version, provided the project is configured to use that registry.
+
+#### As a MetaMask engineer
```

Comment:
This is a little bit fuzzy (maybe intentional? If so that's great and I think it's appropriate to keep)

Flavors:
  - If we want to put a clear line (risking being kind of bureaucratic about it): `As a MetaMask organization member`
  - Implied might work: `As a MetaMask member`
  - To make it more open: `As a MetaMask developer`
  - To make that more inviting: `As a MetaMask contributor`

Reply:
Good point, I was thinking of a short way of saying the first thing. I like "As a MetaMask contributor".

Reply:
Updated in [b321b41](https://github.com/MetaMask/controllers/pull/993/commits/b321b413beb5d24ddd7cbea31e3e27bbaa93c80b).